Sql

romanovsavelij opened the pull request.

DGSB095 committed:
Added lsat changes

Last version

DGSB095 committed:
Create README.md

DGSB095 committed:
Update task.py

DGSB095 committed:
Update task.py

DGSB095 committed:
Add frontend structure

DGSB095 committed:
IDK fronend + backend

DGSB095 committed:
bebebeb

DGSB095 committed:
fineshed header

DGSB095 committed:
header v2

DGSB095 committed:
Update style.css

DGSB095 committed:
header + github

DGSB095 committed:
finished SQL

DGSB095 committed:
k

DGSB095 committed:
Added FASTAPI functions

romanovsavelij commented on task.py:
```diff
@@ -1,58 +1,177 @@
+from pydantic import BaseModel
+from typing import List, Optional
+
+app = FastAPI()
```
Лучше не смешивать разные уровни абстракции, тем более все в один файл засовывать. API отдельно, бизнес логика отдельно, работа с базой отдельно

DGSB095 replied:
- added *api.py* file
- added *models.py* file
- renamed and changed *task.py* to *task_manager.py*
- changed *script.js*

romanovsavelij commented on task.py:
```diff
@@ -1,58 +1,177 @@
+                            task.should_repeat, task.delete_on_complete))
+
+        tmanager_conn.commit()
+        tmanager_conn.close()
```
полезно использовать контекстные менеджеры в python для закрытия соединения. чтобы если посредине кода функции вылетит исключение, соединение все равно закрылось

romanovsavelij commented on frontend/index.html:
```diff
@@ -0,0 +1,46 @@
+</head>
+<body>
+    <header>
+        <nav class="navbar">
```
Код фронтенда тоже важно качественно оформлять с ростом проекта. Чтобы компоненты можно было переиспользовать. Например, вынести navbar в отдельную React компоненту в отдельном файле. В идеале index должен состоять из нескольких строк, реализация так не торчать наружу

romanovsavelij commented on README.md:
```diff
@@ -0,0 +1 @@
+# LifeKeeper
```
В readme полезно написать про то, как разворачивать ваше приложение. Как запустить бекенд, фронтенд, чтобы другой разработчик мог все то же самое проделать у себя

romanovsavelij commented on task.py:
```diff
@@ -1,58 +1,177 @@
+        cursor = tmanager_conn.cursor()
+
+        # Clear existing data
+        cursor.execute("DELETE FROM Sections")
```
Не забывай писать тесты. На базу данных кстати тоже можно и нужно писать тесты. Для этого в тестах должна подниматься тестовая база данных локально, все запросы должны идти в нее. Так разрабочики тестируют, что работа с базой правильно реализована

romanovsavelij replied:
новое сообщение

DGSB095 committed:
Frontend 1 ????

DGSB095 commented:
sadasd

DGSB095 committed:
Project Levels

DGSB095 committed:
Closed nearly all requests

DGSB095 committed:
Docker Upd

- Can run app with Docker now
- Updated README.md
- SQL came back
- React app now exists
- Home and TasksAndSections pages added
- Project structure has changed
- Updated api.py
- Deleted main.py
- etc.

DGSB095 committed:
GOD HELP ME PLS

DGSB095 committed:
ITS ADDING SECTIONS

DGSB095 committed:
taskdetails in progress

DGSB095 committed:
asdasd

DGSB095 committed:
Update task_manager.py
